06_puppy/pfannerj/types.go

```diff
@@ -0,0 +1,17 @@
+package main
+
+//Puppy encapsulates important details about the Puppy including ID, Breed, Colour and Value
+type Puppy struct {
+	ID     uint32  `json:"id"`
+	Breed  string  `json:"breed"`
+	Colour string  `json:"colour"`
+	Value  float64 `json:"value"`
+}
+
+// Storer defines standard CRUD operations for Puppy
+type Storer interface {
+	CreatePuppy(*Puppy) (uint32, error)
+	ReadPuppy(ID uint32) (*Puppy, error)
+	UpdatePuppy(ID uint32, puppy *Puppy) (uint32, error)
+	DeletePuppy(ID uint32) (bool, error)
```

Comment:
as mentioned before save yourself time by returning only the error with this function